### DISCUSSION TITLE
archive log files in main and renderer process after certain size

### DISCUSSION THREAD

**Discussion by arunbansal-cpi** [megahertz/electron-log#359] [Q&A] [answered]
I want to have log rotation of files after certain size to a given number(say 5) in main and renderer processes.
I have overidden archivelog function as follow

```js
const logger = require('electron-log');
const glob = require('glob');
const no_of_files = 5;

logger.transports.file.maxSize = 512;
logger.transports.file.resolvePath = () => path.join('/tmp', `${app.name}.log`);
logger.transports.file.archiveLog = (file) => {
  const oldPath = file.toString();
  const inf = path.parse(oldPath);
  let file_count = 0;
  //print(`$(inf.name)`)
  consoel.log(`File rotation :${inf.name}`);
  try {
    glob(`${inf.dir}/${inf.name}`, (err, files) => {
      if (files.length > 0) file_count = files.length - 1;
    });
    console.log(`file_count: ${file_count}`);
    if (file_count >= no_of_files) file_count = 0;
    let new_file_path = path.join(
      inf.dir,
      `${inf.name}.${++file_count}${inf.ext}`
    );
    fs.unlink(new_file_path, (err) => {
      if (err) {
        console.log(`file is not present ${new_file_path}`);
      }
    });
    console.log(`${new_file_path}`);
    fs.renameSync(oldPath, new_file_path);
  } catch (e) {
    logConsole('Could not rotate log', e);
    const quarterOfMaxSize = Math.round(transport.maxSize / 4);
    file.crop(Math.min(quarterOfMaxSize, 256 * 1024));
  }
};
```

Files are not achiving as intended

**Comment by megahertz** [comment, marked as answer]
Sometimes archive log works a bit later then maximum file size exceeds the limit, especially when writing a large log message. Anyway, if it works with the default behavior, then the issue is in your `archiveLog` implementation.

To debug such an issue, I would do the following:
- Rewrite this code to prevent using callbacks, otherwise it's hard to determine the actual code workflow.
- Make a test similar to [that](https://github.com/megahertz/electron-log/blob/master/src/main/transports/file/__specs__/index.spec.js#L27) for your `archiveLog` function.

**Reply by arunbansal-cpi** [reply]
Do I need to override archiveLog function in renderer process as well ? How?

**Reply by megahertz** [reply]
Yes if you still use electron-log@v4. In v5 you only need to do that in the main process.
